### ISSUE TITLE
Capitalizing the German letter ß gives SS instead of ẞ

### ISSUE THREAD

**Issue by septsea** [VSCodeVim/Vim#8009]
**Describe the bug**
Capitalizing the German letter *ß* gives *SS* instead of *ẞ*.

**To Reproduce**

1.  Type any text that contains *ß*. (For example, *Gauß und Weierstraß*.)
2.  Select the text.
3.  Press `gU`.
4.  *Gauß und Weierstraß* is converted to *GAUSS UND WEIERSTRASS*.

**Expected behavior**
Since there is already the uppercase version of *ß* (https://glyphsapp.com/learn/localize-your-font-german-capital-sharp-s), I expect the result to be *GAUẞ UND WEIERSTRAẞ*.

**Screenshots**
If remapping-related, please attach log output: https://github.com/VSCodeVim/Vim#debugging-remappings.
![Animation](https://user-images.githubusercontent.com/57337516/194034777-c6a113cb-0efb-43fe-ae57-0fb171b1c06b.gif)

**Environment (please complete the following information):**

- Extension (VsCodeVim) version: `1.24.1`
- VSCode version: `1.71.2`
- OS: `Windows_NT x64 10.0.25211`

**Additional context**

**Comment by J-Fields** [comment]
Hmm... looks like in my locale, at least, `'ß'.toUpperCase()` and `'ß'.toLocaleUpperCase()` both return `'SS'`. This could be a locale issue though (this stuff is definitely not my wheelhouse).

@septsea Can you please confirm whether the `g~` (toggle case) command works correctly on your machine with `'ß'`? I noticed `gU` and `gu` use the locale-agnostic methods while `g~` uses the locale-sensitive methods.

**Comment by septsea** [comment]
The `g~` commands converts `ß` to `SS`. @J-Fields

**Comment by J-Fields** [comment]
Yeah, my testing suggests this is an issue with javascript's `toLocaleUppercase` (at least in common implementations) even with the German locale.

**Comment by septsea** [comment]
I did some experiments.

- `"Möbelträgerfüße".upper()` in Python returns `MÖBELTRÄGERFÜSSE`.
- `"Möbelträgerfüße".toUpperCase()` in Java returns `MÖBELTRÄGERFÜSSE`.
- `"Möbelträgerfüße".upcase` in Ruby returns `MÖBELTRÄGERFÜSSE`.
- `"Möbelträgerfüße".uppercased()` in Swift returns `MÖBELTRÄGERFÜSSE`.
- `"Möbelträgerfüße".to_uppercase()` in Rust returns `MÖBELTRÄGERFÜSSE`.
- `"Möbelträgerfüße".ToUpper()` in C# returns `MÖBELTRÄGERFÜßE` (instead of `MÖBELTRÄGERFÜẞE`).
- `strings.ToUpper("Möbelträgerfüße")` in Go returns `MÖBELTRÄGERFÜßE` (instead of `MÖBELTRÄGERFÜẞE`).

Well. `ẞ` was nonexistent or nonstandard when `toUppercase` (or `toLocaleUppercase`, or their counterparts in other programming languages) was created.

**Comment by lordofscripts** [comment]
Last year I ran into a description somewhere that there was indeed such a problem with THAT German character. To me it was (and is) a problem because I am working on an application where that translation should return ONE character, not two ("ss").

And now in another application with a TabulaRecta things are going amiss thanks to THAT specific character.
